polkadot/node/core/pvf/tests/it/main.rs

```diff
@@ -317,3 +325,65 @@ async fn deleting_prepared_artifact_does_not_dispute() {
 		r => panic!("{:?}", r),
 	}
 }
+
+// This test checks if the adder parachain runtime can be prepared with 10Mb preparation memory
+// limit enforced. At the moment of writing, the limit if far enough to prepare the PVF. If it
+// starts failing, either Wasmtime version has changed, or the PVF code itself has changed, and
+// more memory is required now. Multi-threaded preparation, if ever enabled, may also affect
+// memory consumption.
+#[cfg(feature = "tracking-allocator")]
+#[tokio::test]
+async fn prechecking_within_memory_limits() {
+	let host = TestHost::new();
+	let parent_head = HeadData { number: 0, parent_hash: [0; 32], post_state: hash_state(0) };
+	let block_data = BlockData { state: 0, add: 512 };
+	let result = host
+		.validate_candidate(
```

Comment:
This runs not just `prechecking` but also validation as well. Is this intentional? You can use the new `TestHost::precheck_pvf` method I added [here](https://github.com/paritytech/polkadot-sdk/pull/1373/commits/a696d406b0ada3eb8664a1eff5e7410e11e81b35).

Reply:
Ah great, I'm going to steal your code then :grin: